packages/aws-rfdk/lib/deadline/lib/worker-configuration.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+
+import {
+  OperatingSystemType,
+} from '@aws-cdk/aws-ec2';
+import {
+  IGrantable,
+} from '@aws-cdk/aws-iam';
+import {
+  Construct,
+  Duration,
+} from '@aws-cdk/core';
+import {
+  CloudWatchAgent,
+  CloudWatchConfigBuilder,
+  IScriptHost,
+  LogGroupFactory,
+  LogGroupFactoryProps,
+  ScriptAsset,
+} from '../../core';
+import { Version } from './version';
+
+/**
+ * Interface for Deadline clients that can be configured via the ClientConfiguration
+ * helper class.
+ */
+export interface IConfigurableWorker extends IScriptHost, IGrantable {
+}
+
+/**
+ * The Deadline Worker settings that can be configured via the configureWorkerSettings method
+ * of the WorkerConfiguration helper class.
+ */
+export interface WorkerSettings {
+  /**
+   * Deadline groups these workers needs to be assigned to. The group is
+   * created if it does not already exist.
+   *
+   * @default - Worker is not assigned to any group
+   */
+  readonly groups?: string[];
+
+  /**
+   * Deadline pools these workers needs to be assigned to. The pool is created
+   * if it does not already exist.
+   *
+   * @default - Worker is not assigned to any pool.
+   */
+  readonly pools?: string[];
+
+  /**
+   * Deadline region these workers needs to be assigned to.
+   *
+   * @default - Worker is not assigned to any region
+   */
+  readonly region?: string;
+}
+
+/**
+ * This is a helper construct for configuring Deadline Workers to connect to a RenderQueue, send their
+ * log files to CloudWatch, and similar common actions.
+ *
+ * Security Considerations
+ * ------------------------
+ * - The instances configured by this construct will download and run scripts from your CDK bootstrap bucket when that instance
+ *   is launched. You must limit write access to your CDK bootstrap bucket to prevent an attacker from modifying the actions
+ *   performed by these scripts. We strongly recommend that you either enable Amazon S3 server access logging on your CDK
+ *   bootstrap bucket, or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ */
+export class WorkerConfiguration extends Construct {
```

Comment:
Should we rename this construct to `ClientConfiguration`? The construct looks generic enough to configure any deadline client eg. Monitor.

Reply:
The generic client configuration is the render queue's client connect methods. The worker configuration script has some very worker-specific stuff in it like getting the Launcher to start a Worker.